baserun/annotation.py

```diff
@@ -1,100 +1,80 @@
 class Annotation:
-    completion_id: str
+    completion_id: Optional[str]
     span: Span
-    input_variables: list[InputVariable]
-    logs: list[Log]
-    checks: list[Check]
-    feedback_list: list[Feedback]
+    input_variables: List[InputVariable]
+    logs: List[Log]
+    checks: List[Check]
+    feedback_list: List[Feedback]
 
-    def __init__(self, completion_id: Optional[str] = None, run: Optional[Run] = None):
+    def __init__(self, completion_id: Optional[str] = None, run: Optional[Run] = None) -> None:
         self.run = run or Baserun.get_or_create_current_run()
         self.input_variables = []
         self.logs = []
         self.checks = []
         self.feedback_list = []
 
         if span := self.try_get_span():
+            # TODO: I don't think it's used anywhere
             self.span = span
 
-        if completion_id:
-            self.completion_id = completion_id
+        # we can't conditionally add this attribute like before because you might get AttributeError then
+        self.completion_id = completion_id
 
+    # Annotation initializer from Baserun class. I felt like it didn't belong there
     @classmethod
-    def annotate(cls, completion: Union[None, ChatCompletion, Stream[ChatCompletionChunk]] = None):
+    def exported_annotate(
+        cls, completion_id: Optional[str] = None, run: Optional[Run] = None, trace: Optional[Run] = None
+    ) -> "Annotation":
+        """Capture annotations for a particular run and/or completion. the `trace` kwarg here is simply an alias"""
+
+        return cls(completion_id=completion_id, run=run or trace)
+
+    @classmethod
+    def annotate(cls, completion: Union[None, ChatCompletion, Stream[ChatCompletionChunk]] = None) -> "Annotation":
         if isinstance(completion, ChatCompletion):
             completion_id = completion.id
             return cls(completion_id=completion_id)
         else:
             return cls()
 
-    @classmethod
-    @asynccontextmanager
-    async def aanotate(cls, completion: Union[None, ChatCompletion, Stream[ChatCompletionChunk]] = None):
-        if not Baserun._initialized:
-            yield
-
-        annotation = cls.annotate(completion=completion)
-        try:
-            yield annotation
-        finally:
-            try:
-                await annotation.asubmit()
-            except BaseException as e:
-                logger.warning(f"Could not submit annotation to baserun: {e}")
-
-    @classmethod
-    @contextmanager
-    def with_annotation(cls, completion: Union[None, ChatCompletion, Stream[ChatCompletionChunk]] = None):
-        if not Baserun._initialized:
-            yield
-
-        annotation = cls.annotate(completion=completion)
-        try:
-            yield annotation
-        finally:
-            try:
-                annotation.submit()
-            except BaseException as e:
-                logger.warning(f"Could not submit annotation to baserun: {e}")
-
```

Comment:
this it's not used anywhere nor is it documented

Reply:
Or, if you're just talking about the context manager, I would still like to keep it